File: androidApp/src/main/java/com/netguru/retail/android/mocks/ShopServiceMock.kt
package com.netguru.retail.android.mocks

import com.netguru.commondomain.shop.ShopService
import kotlinx.coroutines.Job

class ShopServiceMock : ShopService {
    override fun loadData(): Job = Job()

    override fun openProductDetails(productId: String): Job = Job()

    override fun searchProduct(productName: String): Job = Job()
}

File: iosApp/iosApp/Source Files/Common/Mocks/ShopServiceMock.swift
//
//  ShopServiceMock.swift
//  R&D KMM Retail
//

import Foundation
import common

class ShopServiceMock: ShopService {

    func loadData() -> Kotlinx_coroutines_coreJob {
        MockProvider.shared.job
    }

    func openProductDetails(productId: String) -> Kotlinx_coroutines_coreJob {
        MockProvider.shared.job
    }

    func searchProduct(productName: String) -> Kotlinx_coroutines_coreJob {
        MockProvider.shared.job
    }
}
